aa ICTRAINING 366 Django Tutorial

adil-atri opened the pull request:
Created a new Django project and added the Polls app.
Part 1 of the tutorial completed.

[ICTRAINING-367]

adil-atri committed:
feat(Django Tutorial): Project init

Created new Django project

[ICTRAINING-367]

adil-atri committed:
feat(Django Tutorial): Polls App

Created the Polls application and added an index view.

[ICTRAINING-367]

adil-atri committed:
feat(Django Tutorial): DB Setup

Database setup

[ICTRAINING-368]

xiaoleATRI commented on mysite/settings.py:
```diff
@@ -0,0 +1,120 @@
+# Database
+# https://docs.djangoproject.com/en/3.0/ref/settings/#databases
+
+DATABASES = {
```
```suggestion
# your local settings
DATABASES = {
'default': {
        'ENGINE': 'django.db.backends.postgresql_psycopg2',
        'NAME': 'your database name',
        'USER': 'your username to access the postgres database',
        'PASSWORD': 'your password to access the postgres database',
        'HOST': 'localhost',
        'PORT': 5432,
}}
```

xiaoleATRI reviewed (approved):
Approved

adil-atri committed:
feat(Django Tutorial): Created models

Added question and choice models

 [ICTRAINING-368]

adil-atri committed:
feat(Django Tutorial): Add Question to Admin

Added Quesiton objects to admin interface

[ICTRAINING-368]

adil-atri commented:
Set up the database and created models.
Added the question model to the admin interface.
Part 2 of the tutorial completed.

[ICTRAINING-368]

xiaoleATRI reviewed (approved):
Approved

adil-atri committed:
feat(Django Tutorial): Created Polls views

Made modifications to the Polls application:
-modified index view
-created detail, results and vote views
-added index and detail templates

[ICTRAINING-369]

adil-atri commented:
Created views for the Polls app.
Part 3 of the tutorial completed.

[ICTRAINING-369]

adil-atri committed:
feat(Django Tutorial): Modified Views

Modified results and vote views
Modified detail template by adding a form
Added results template

[ICTRAINING-370]

adil-atri committed:
feat(Django Tutorial): Generic Views

Modified index view to use generic ListView
Modified detail and results views to use generic DetailView

[ICTRAINING-370]

adil-atri commented:
Modified existing views and used generic views.
Part 4 of the tutorial completed.

[ICTRAINING-370]

xiaoleATRI commented on polls/views.py:
```diff
@@ -0,0 +1,40 @@
+            'error_message': "You didn't select a choice"
+        })
+    else:
+        selected_choice.votes = F('votes') + 1
```
Question:
why do you use query expression here instead of pure accumulation?
`selected_choice.votes +=1`

adil-atri replied:
In case two users vote at the same time. The F() method takes care of that issue according to documentation to have the database update that value instead of Python

xiaoleATRI reviewed (approved):
Approved

adil-atri committed:
feat(Django Tutorial): Method Testing

Added tests for the was_published_recently() method

[ICTRAINING-371]

adil-atri committed:
feat(Django Tutorial): View Testing

Added tests for the Question index and detail views

[ICTRAINING-371]

adil-atri commented:
Added method and view tests.
Part 5 of the tutorial completed.

[ICTRAINING-371]

adil-atri committed:
feat(Django Tutorial): Static files

Added styling and background image to index template

[ICTRAINING-372]

adil-atri commented:
Added static files.
Part 6 of the tutorial completed.

[ICTRAINING-372]

adil-atri committed:
feat(Django Tutorial): Admin forms

Customized question admin forms.
Changed site name.

[ICTRAINING-373]

adil-atri commented:
Customization of admin forms.
Part 7 of the tutorial completed.

[ICTRAINING-373]

adil-atri committed:
feat(Django Tutorial): Heroku Deployment

Setting up heroku deploy

[ICTRAINING-374]

adil-atri committed:
feat(Django Tutorial): Heroku Deployment

Adding heroku-django

[ICTRAINING-374]
